### PR TITLE
feat(stats): account for prior (pre-app) dives & time in lifetime totals

### LINKED ISSUE #331: Dives/dive time prior to logging accounted for in stats
Provide a way to tell Submersion I have x amount of dives/time that doesn't exist in logs.

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

Lets each diver record diving they did **before** they started using Submersion, so the Stats page reflects a real career instead of only app-logged dives. Closes #331.

- **Data:** three nullable columns on `divers` (`prior_dive_count`, `prior_dive_time_seconds`, `diving_since`), added in a PRAGMA-guarded schema **v87** migration. They flow through the `Diver` entity + repository and sync automatically via the row's existing HLC and the generic `toJson` export (no sync-layer changes needed).
- **Logic:** a pure, unit-tested `CareerTotals` model combines logged statistics with the active diver's prior values.
- **Display:** Total Dives and Total Time now show the **combined** total with a small `logged + prior` breakdown, plus a "Diving since YYYY" line. Reconciliation shows the earlier of the entered year vs. the first logged dive.
- **Entry:** a "Prior Experience" section on the diver edit form (dive count, hours + minutes, year picker).
- **Scope:** only the two headline totals are offset — averages, deepest dive, and charts stay logged-only (no per-dive data exists for the paper-log era). The page is **byte-for-byte unchanged** for divers who don't use the feature.
- **i18n:** new strings translated into all 10 non-English locales.

## Test Plan

- [x] Migration v86→v87 adds columns and preserves existing rows
- [x] `Diver` entity + repository round-trip the new fields
- [x] `CareerTotals` combine logic (logged+prior, partial, negatives, diving-since reconciliation)
- [x] Edit-form save wiring (widget test captures the saved `Diver`)
- [x] Stats page shows combined totals + breakdown; logged-only with no prior experience (widget test)
- [x] Statistics (72), divers + database (192) suites green; whole-project `flutter analyze` clean
- [ ] Manual device/sim: enter prior experience → verify Stats; clear it → verify page returns to today's look
- [x] Native-speaker spot-check of `he` / `ar` / `zh` / `hu` translations

### CODE REVIEW

**Comment by ericgriffin** [comment]
Addressed all six review comments (pushed in 33c8508):

1. **Time breakdown dropped minutes / showed "0h prior"** — the logged/prior time breakdown now formats as `Xh Ym` via a shared helper, so it preserves minutes and always reconciles with the combined headline.
2–5. **`divingSince` stored as epoch-millis → cross-timezone year shift** — now stored as a bare **year integer** (`diver.divingSince?.year`) and read back as `DateTime(year)`; the column comment is corrected. Sync exports the raw year, so there's no timezone drift on sync/restore.
6. **Empty-state hid prior experience** — `CareerTotals` is now computed *before* the empty-state check, so a diver with prior experience but zero app-logged dives sees their career total instead of the empty state.

Also added tests for the edit-page load/validators/year-picker and the zero-logged career path.

**Patch coverage: 81% → 98%.** The only remaining uncovered lines are the three Drift column getters on `divers`, which are shadowed by generated `late final` fields and so are not executable at runtime.

File: lib/features/statistics/presentation/pages/statistics_overview_page.dart [hunk 1]
```diff
@@ -122,13 +151,25 @@ class _AggregateGrid extends StatelessWidget {
...
         icon: Icons.timer,
         label: 'Total Time',
-        value: stats.totalTimeFormatted,
+        value: career.combinedTimeFormatted,
+        subtitle: career.hasPriorTime
+            ? context.l10n.statistics_priorBreakdown(
+                '${career.loggedHours}h',
+                '${career.priorHours}h',
+              )
+            : null,
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/statistics/presentation/pages/statistics_overview_page.dart:182]: Addressed in 33c8508: the logged/prior time breakdown now formats as `Xh Ym` (shared `_formatHm` helper, exposed as `loggedTimeFormatted`/`priorTimeFormatted`), so it preserves minutes and reconciles with the combined headline.

File: lib/core/database/database.dart [hunk 2]
```diff
@@ -44,6 +44,12 @@ class Divers extends Table {
   /// (nullable: rows written before HLC rollout fall back to updatedAt).
   TextColumn get hlc => text().nullable()();
 
+  // Prior dive experience (issue #331): per-diver lifetime offsets for dives
+  // logged before the diver started using Submersion. Null = none.
+  IntColumn get priorDiveCount => integer().nullable()();
+  IntColumn get priorDiveTimeSeconds => integer().nullable()();
+  IntColumn get divingSince => integer().nullable()(); // Unix ms timestamp
+
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/core/database/database.dart:52]: Addressed in 33c8508: `divingSince` is now stored as a bare year integer and the column comment is corrected. Sync exports the raw year, so there is no cross-timezone year shift.

File: lib/features/divers/data/repositories/diver_repository.dart [hunk 3]
```diff
@@ -141,6 +141,9 @@ class DiverRepository {
               isDefault: Value(diver.isDefault),
               createdAt: Value(now.millisecondsSinceEpoch),
               updatedAt: Value(now.millisecondsSinceEpoch),
+              priorDiveCount: Value(diver.priorDiveCount),
+              priorDiveTimeSeconds: Value(diver.priorDiveTimeSeconds),
+              divingSince: Value(diver.divingSince?.millisecondsSinceEpoch),
             ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/divers/data/repositories/diver_repository.dart:147]: Addressed in 33c8508: `createDiver` now writes `diver.divingSince?.year`.

File: lib/features/divers/data/repositories/diver_repository.dart [hunk 4]
```diff
@@ -199,6 +202,9 @@ class DiverRepository {
           notes: Value(diver.notes),
           isDefault: Value(diver.isDefault),
           updatedAt: Value(now),
+          priorDiveCount: Value(diver.priorDiveCount),
+          priorDiveTimeSeconds: Value(diver.priorDiveTimeSeconds),
+          divingSince: Value(diver.divingSince?.millisecondsSinceEpoch),
         ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/divers/data/repositories/diver_repository.dart:208]: Addressed in 33c8508: `updateDiver` now writes `diver.divingSince?.year`.

File: lib/features/divers/data/repositories/diver_repository.dart [hunk 5]
```diff
@@ -658,6 +664,11 @@ class DiverRepository {
       isDefault: row.isDefault,
       createdAt: DateTime.fromMillisecondsSinceEpoch(row.createdAt),
       updatedAt: DateTime.fromMillisecondsSinceEpoch(row.updatedAt),
+      priorDiveCount: row.priorDiveCount,
+      priorDiveTimeSeconds: row.priorDiveTimeSeconds,
+      divingSince: row.divingSince != null
+          ? DateTime.fromMillisecondsSinceEpoch(row.divingSince!)
+          : null,
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/divers/data/repositories/diver_repository.dart]: Addressed in 33c8508: now reads the stored year via `DateTime(row.divingSince!)`.

File: lib/features/statistics/presentation/pages/statistics_overview_page.dart [hunk 6]
```diff
@@ -48,14 +50,36 @@ class _OverviewBody extends ConsumerWidget {
...
+      loggedDives: stats.totalDives,
+      loggedTimeSeconds: stats.totalTimeSeconds,
+      firstLoggedDive: stats.firstDiveDate,
+      priorDives: diver?.priorDiveCount,
+      priorTimeSeconds: diver?.priorDiveTimeSeconds,
+      divingSince: diver?.divingSince,
+    );
     final recordsAsync = ref.watch(diveRecordsProvider);
 
     return SingleChildScrollView(
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/statistics/presentation/pages/statistics_overview_page.dart]: Addressed in 33c8508: `CareerTotals` is computed before the empty-state guard, so a diver with prior experience but zero app-logged dives sees their combined career total instead of the empty state.

File: lib/features/statistics/presentation/pages/statistics_overview_page.dart [hunk 7]
```diff
@@ -42,7 +44,19 @@ class _OverviewBody extends ConsumerWidget {
...
+      priorDives: diver?.priorDiveCount,
+      priorTimeSeconds: diver?.priorDiveTimeSeconds,
+      divingSince: diver?.divingSince,
+    );
+    // Show the career view when the diver has prior experience even if nothing
+    // is logged in-app yet -- otherwise the feature's target user (a long
+    // pre-app career, no logged dives) would only ever see the empty state.
+    if (stats.totalDives == 0 && !career.hasPriorExperience) {
       return const _EmptyState();
     }
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/statistics/presentation/pages/statistics_overview_page.dart]: Addressed in 37701ce: `_OverviewBody` now waits for `currentDiverProvider` to resolve before the empty-state check (shows a loader when there are no logged dives and the diver is still loading), so a diver with prior experience never flashes the empty state. Added a loading-state widget test.

File: lib/features/divers/presentation/pages/diver_edit_page.dart [hunk 8]
```diff
@@ -741,6 +769,110 @@ class _DiverEditPageState extends ConsumerState<DiverEditPage> {
...
+                    : context.l10n.divers_edit_divingSinceNotSet,
+              ),
+              trailing: _divingSince != null
+                  ? IconButton(
+                      icon: const Icon(Icons.clear),
+                      onPressed: () => setState(() {
+                        _divingSince = null;
+                        _hasChanges = true;
+                      }),
+                    )
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/divers/presentation/pages/diver_edit_page.dart:867]: Addressed in 37701ce: added a tooltip to the diving-since clear button via a new `divers_edit_clearDivingSinceTooltip` string (translated into all locales), matching the medical/insurance clear buttons.

File: docs/superpowers/specs/2026-06-15-prior-dive-experience-design.md [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+
+### 1. Data model & storage
+
+Add three nullable columns to the `Divers` table
+(`lib/core/database/database.dart`):
+
+- `priorDiveCount` — `IntColumn`, nullable.
+- `priorDiveTimeSeconds` — `IntColumn`, nullable. Stored in seconds to match the
+  existing `totalTimeSeconds`.
+- `divingSince` — `DateTimeColumn`, nullable. Stored as a date; entered as a year.
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply docs/superpowers/specs/2026-06-15-prior-dive-experience-design.md]: Addressed in 37701ce: the design spec now documents `divingSince` as an `IntColumn` storing the year integer (not a `DateTimeColumn`).

File: lib/core/database/database.dart [hunk 10]
```diff
@@ -1613,7 +1619,7 @@ class AppDatabase extends _$AppDatabase {
 
   /// The current schema version as a static constant so that pre-open checks
   /// (e.g. version-mismatch guard) can reference it without an instance.
-  static const int currentSchemaVersion = 86;
+  static const int currentSchemaVersion = 87;
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/core/database/database.dart:1622]: Addressed: PR description updated to reference schema **v87** and the v86->v87 migration, matching the code after the incremental-sync merge renumbered it from v84.
